### PR TITLE
Register `cortex_discarded_requests_total`

### PR DESCRIPTION
**Pull request by LeviHarrison**
#### What this PR does

Registers `cortex_discarded_requests_total` as it wasn't previously registered.

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by LeviHarrison** [comment]
Added a changelog entry because the metric is technically user-facing I guess?

**Review by treid314 — APPROVED** [review]
Good catch! LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 ### Grafana Mimir
 
 * [BUGFIX] Fix reporting of tracing spans from PromQL engine. #2707
+* [BUGFIX] Register `cortex_discarded_requests_total`, which previosly was not registered and therefore not exported. #2712
```
Comment (56quarters) [inline CHANGELOG.md:8]:
```suggestion
* [BUGFIX] Distributor: Register `cortex_discarded_requests_total` metric, which previously was not registered and therefore not exported. #2712
```

**Review by 56quarters — APPROVED** [review]
LGTM, small fix.

**Comment by 56quarters** [comment]
Looks good, just need to resolve a merge conflict

**Comment by LeviHarrison** [comment]
Can't fix them fast enough!

**Pull request merged**